### PR TITLE
Feature/search favorites script

### PR DESCRIPTION
**Pull request by cagallo**
#### What's this PR do?

- Fix API catch issue
- Fix argument for display recipes in the searchallrecipes function
- delete scripts and style from html (not needed)

#### Where should the reviewer start?

- scripts.js

#### How should this be manually tested?

- npm start

#### Any background context you want to provide?

- N/A

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -2,7 +2,6 @@
 <html lang="en">
 <head>
   <meta charset="UTF-8">
-  <link rel="stylesheet" href="styles.css">
```
Comment (anewb87) [inline src/index.html:5]: Thanks be to Hannah for helping us out with error messages due to this line and the one at the bottom of the file.

File: src/scripts.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ Promise.all([userAPI, ingredientAPI, recipeAPI])
     })
     recipeRepository = new RecipeRepository(recipeList)
     displayRecipes(recipeRepository.recipeData)
-  }).catch(console.log('NO SOUP FOR YOU!'))
+  }).catch(error => console.log(error))
```
Comment (aliroemhildt) [inline src/scripts.js:29]: great catch!

Comment (anewb87) [inline src/scripts.js:29]: We believe this fixed the issue.

File: src/scripts.js [hunk 3]
```diff
@@ -248,7 +248,7 @@ function createButtonEventListeners() {
 function searchAllRecipes() {
   const searchName = document.getElementById('searchInput').value;
   recipeRepository.filterByNameOrIng(searchName, ingredientsData);
-  displayRecipes();
+  displayRecipes(recipeRepository.recipesToShow);
```
Comment (aliroemhildt) [inline src/scripts.js:251]: woohoo, glad it was an easy fix!

**Review by aliroemhildt — APPROVED** [review]

File: src/scripts.js (hunk 3 above)
Comment (anewb87) [inline src/scripts.js:251]: Fixed what we broke- search bar works again.

**Review by anewb87 — APPROVED** [review]
A few small, quick changes that make a difference.

**Review by kammon10 — APPROVED** [review]

**Comment by aliroemhildt** [comment]
**Questions for Kristen:**

We got our API fetch call set up and functioning! We just wanted to see if you could take a look and let us know how it looks to you

Specific questions:
- Catch statement: We are wondering your advice on what an appropriate catch statement is, right now we are just console logging but are not sure if there is something else that is considered "best practice"
- Promise statement: Does it seem okay how we have ours set up with all of our global variable instantiations and page load function inside the promise? Or would it be better to move these outside of the promise and only keep the data objects that the promise is returning actually inside the promise? 
- fetch get calls are in apiCalls.js, the Promise is in lines 17-29 of scripts.js

**Comment by kristenmb** [comment]
I've included a lot of info here but want to preface it with this: What you have looks GREAT. Its functional and it's your first time using fetch in a project so don't stress too much about it being perfect! If there is too much stress/pressure around getting a few things tweaked here, don't do it. Seriously. Its working as is and thats the most important bit!

That being said here are some of my thoughts!:

### RE Catch statements: 

I think its important to understand the anatomy of a fetch call -- 

1. You call the fetch and have a promise waiting to be fulfilled.
2. If the promise is successful, the fetch moves onto a `.then` 
3. The first `.then ` is used to parse the data from json and make it readable and usable in javascript
4. If there is a second .`then` the fetch call moves into that. This is where you tell the code what to do with your newly parsed data (store it in a variable, use it in some functionality, etc)
5. If the promise returns unsuccessfully, the fetch moves into the `.catch`, which handles any bad responses and is used for error handling.

If you think back to projects like Intention Timer where you had a lot of error handling, the `catch` is a good place to trigger errors. in the same way that the .then can set the returned data to a variable, your .catch could potentially create/show an error message on the DOM indicating that something went wrong. Think through a few ways you could have the `.catch` so some error handling for you - maybe it calls a `displayError` function that has a message prompting a user to try again later or refresh the page. 

### RE: Promise.all

**Again starting with some background:**
Every promise is an async function - so every fetch call runs to the side or separately from your `script.js ` file. The reason we would use a Promise.all function is because your main page utilizes data from all three fetch calls (userData, Ingredients, and Recipes) and we want to be sure we have ALL of this information before we load the page. If you run the three fetch calls separately one may return before another and make the logic that you currently have in your promise.all (namely randomly choosing a user) a bit more complicated.

With the Promise.all call, the function will wait for EVERY fetch in the argument array to be fulfilled before it allows the functions to move on to the .then or .catch  methods, meaning you have EVERYTHING you need to choose a random user and populate your page with data.

There are a lot of ways to go about this process, as with any code, but I generally lean towards using whatever feels like its the easiest to read. For example, I see in your apiCalls file you have all your second .then  calls commented out, but if you use those, your Promise.all can be simplified to this:

```
Promise.all([userAPI, ingredientAPI, recipeAPI])
  .then(data => {
    userData = data[0];
    ingredientsData = data[1];
    recipeData = data[2];
    userIndex = getRandomIndex(usersData);
    currentUser = new User(usersData[userIndex], ingredientsData);
    recipeList = recipeData.map(recipe => {
      return new Recipe(recipe);
    })
    recipeRepository = new RecipeRepository(recipeList)
    displayRecipes(recipeRepository.recipeData)
  })
  .catch(error => console.log(error))
```

which feels a bit more readable to me.

If you wanted to clean it up a bit more, I would say you could split the `userIndex` and `currentUser` assignment out into a separate function pretty easily. maybe called `setCurrentUser` and you could do something similar with the recipe instantiations -- because you've stored the fetched data globally you can access it from any function. 

```
Promise.all([getAllUsers(), ingredientAPI, recipeAPI])
  .then(data => {
    userData = data[0];
    ingredientsData = data[1];
    recipeData = data[2];
    setCurrentUser();
    createRecipeRepo();
  }).catch(error => console.log(error))

```

I also think its best to avoid having functions automatically call on page load. I would move your promise.all into a function and create a page load event listener.


### Misc
Only other piece I noticed was that you have your fetch calls stored as variables. While this Is fine and clearly works, I prefer to having my fetch calls return from a function, for example

```
const getAllUsers = () => {
  return fetch("https://what-s-cookin-starter-kit.herokuapp.com/api/v1/users")
  .then(response => response.json())
  .then(data => data.usersData)
}
```

This could then be exported from your api calls, imported into the script file and the Promise.all would  look like this:
`Promise.all([getAllUsers(), ingredientAPI, recipeAPI]) `(and of course. I would do it with the other fetch calls as well)

Same effect and same result,  but good practice for when you have fetch calls that you might need to pass parameters into, and this also allows you to execute a `.then` or `.catch` in your script file. For example, in the error handling we were talking about earlier - if you imported `getAllUsers()` into your script file and for this example lets say we're using it outside of the Promise.all -  you could call the function and add a catch within your script file that handled the error.

EX: 

```
///  SCRIPT.JS FILE ////

const fetchUsers = () => {
  getAllUsers()
    .catch(error => {
      displayErrorMessage(error)
    })
}

const displayErrorMessage = (error) => {
  //some logic to display the error on the DOM
}

```


Again I know this is a lot of information and I'm happy to chat through it with any/all of you. I will reiterate that the way you have everything working right now is great and functional! Reach out with any questions!

**Pull request merged**